test/unit/test_dwarf_info.c

```diff
@@ -0,0 +1,292 @@
+	mu_assert_eq (info->length, 2, "Incorrect number of info compilation units");
+
+	// check header
+	RBinDwarfCompUnit cu = info->comp_units[0];
+	RBinDwarfCompUnitHdr hdr = cu.hdr;
+	mu_assert_eq (hdr.version, 4, "Wrong header information");
+	mu_assert_eq (hdr.length, 0x2c0, "Wrong header information");
+	mu_assert_eq (hdr.is_64bit, false, "Wrong header information");
+	mu_assert_eq (hdr.address_size, 8, "Wrong header information");
+	mu_assert_eq (hdr.abbrev_offset, 0x0, "Wrong header information");
+
+	// check some of the attributes
+	mu_assert_eq (cu.length, 73, "Wrong attribute information");
+	mu_assert_eq (cu.offset, 0x0, "Wrong attribute information");
+
+	int i = 0;
+	mu_assert_eq (cu.dies[i].abbrev_code, 1, "Wrong attribute information");
+	mu_assert_eq (cu.dies[i].length, 7, "Wrong attribute information");
+	mu_assert_eq (cu.dies[i].tag, DW_TAG_compile_unit, "Wrong attribute information");
+	mu_assert_eq (cu.dies[i].attr_values[0].attr_name, DW_AT_producer, "Wrong attribute information");
+	mu_assert_streq (cu.dies[i].attr_values[2].string.content, "../main.cpp", "Wrong attribute information");
+	mu_assert_eq (cu.dies[i].attr_values[6].attr_name, DW_AT_ranges, "Wrong attribute information");
+	mu_assert_eq (cu.dies[i].attr_values[6].reference, 0x0, "Wrong attribute information");
+
+	i++;
+	mu_assert_eq (cu.dies[i].abbrev_code, 2, "Wrong attribute information");
+	i++;
+	mu_assert_eq (cu.dies[i].abbrev_code, 3, "Wrong attribute information");
+	i++;
+	mu_assert_eq (cu.dies[i].abbrev_code, 3, "Wrong attribute information");
+	i++;
+	mu_assert_eq (cu.dies[i].abbrev_code, 3, "Wrong attribute information");
+	i++;
+	mu_assert_eq (cu.dies[i].abbrev_code, 0, "Wrong attribute information");
+	i++;
+	// i == 6
+	mu_assert_eq (cu.dies[i].abbrev_code, 4, "Wrong attribute information");
+	mu_assert_eq (cu.dies[i].attr_values[0].reference, 0x6e, "Wrong attribute information");
+	mu_assert_eq (cu.dies[i].attr_values[1].data, 4, "Wrong attribute information");
+	mu_assert_streq (cu.dies[i].attr_values[2].string.content, "Bird", "Wrong attribute information");
+	mu_assert_eq (cu.dies[i].attr_values[3].data, 8, "Wrong attribute information");
+	mu_assert_eq (cu.dies[i].attr_values[4].data, 1, "Wrong attribute information");
+	mu_assert_eq (cu.dies[i].attr_values[5].data, 9, "Wrong attribute information");
+	i++;
+	mu_assert_eq (cu.dies[i].abbrev_code, 5, "Wrong attribute information");
+	mu_assert_eq (cu.dies[i].tag, DW_TAG_member, "Wrong attribute information");
+	mu_assert_streq (cu.dies[i].attr_values[0].string.content, "_vptr$Bird", "Wrong attribute information");
+	mu_assert_eq (cu.dies[i].attr_values[1].reference, 0xc5, "Wrong attribute information");
+	mu_assert_eq (cu.dies[i].attr_values[2].data, 0, "Wrong attribute information");
+	mu_assert_eq (cu.dies[i].attr_values[3].flag, true, "Wrong attribute information");
+	i++;
+	mu_assert_eq (cu.dies[i].abbrev_code, 6, "Wrong attribute information");
```

Comment:
I think you can fold these repeating elements with a macro, so only number changed, or use the array of numbers.